tools/pre-commit-hooks/check_generated_files.py

```
@@ -0,0 +1,68 @@
+from __future__ import annotations
+
+import os
+import subprocess
+import sys
+
+# Ensure the repo root is on sys.path so warp is importable in the
+# pre-commit isolated virtualenv (which only has additional_dependencies).
+base_path = os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
+sys.path.insert(0, base_path)
+
+import warp  # noqa: E402 — populates builtin_functions via builtins.py
```

Ruff (0.15.1)

[warning] Unused `noqa` directive (non-enabled: `E402`)

Remove unused `noqa` directive

(RUF100)